src/db/seed/index.js

```diff
@@ -0,0 +1,40 @@
+import db from '..';
+import users from './userSeed';
+import motivators from './motivatorSeed';
+import getContents from './motivatorContentSeed';
+import getResults from './motivatorResultSeed';
+import getInputs from './motivatorInputSeed';
+
+const insertSeed = async () => {
+  const t = await db.sequelize.transaction();
+
+  try {
+    const usersCreated = await db.models.user.bulkCreate(users, {
+      transaction: t,
+    });
+
+    const motivatorsCreated = await db.models.motivator.bulkCreate(motivators, {
+      transaction: t,
+    });
+
+    motivatorsCreated.forEach((item) => {
+      // eslint-disable-next-line max-len
+      await db.models.motivatorContent.bulkCreate(getContents(item.id), { transaction: t });
+
+      const resultsCreated = await db.models.motivatorResult.bulkCreate(
+        getResults(item.id, usersCreated[0].id),
+        { transaction: t },
+      );
+
+      resultsCreated.forEach((result) => {
+        await db.models.motivatorResultInput.bulkCreate(getInputs(result.id), { transaction: t });
+      });
+    });
```

Comment:
```suggestion
    await Promise.all(motivatorsCreated.map(async (item) => {
      await db.models.motivatorContent.bulkCreate(getContents(item.id), { transaction: t });

      const resultsCreated = await db.models.motivatorResult.bulkCreate(
        getResults(item.id, usersCreated[0].id),
        { transaction: t },
      );

      await Promise.all(resultsCreated.map(async (result) => {
        await db.models.motivatorResultInput.bulkCreate(getInputs(result.id), { transaction: t });
      }));
    }));
```

A, so here is the catch, why it keeps crashing. The function you are passing to `forEach` uses the await `keyword` but is not marked as async. However just prepending the (anonymous) function definition with `async` would lead you straight into a JavaScript pitfall. 

**`Array.prototype.foreach` is not made for working with async functions**!

It implicitly expects you to pass a synchronous function, even if such is not stated explicitly in the ECMAScript standards. On array prototype functions, you can only use functions that actually return a promise (which is the one implictly returned by async functions). So only `map()`, `flatMap()`, `reduce()` and `reduceRight()` works (if used correctly). `filter()` for example won't work as it relies on the value returned by the passed function and was designed without async code in mind. It would just see the returned promise, recognizes it as *truthy* (as it's not `null`) and accepts the value.

So how to deal with that. There are three ways. Two of them are suitable for processing the values in a serial manner, for for processing parallel. The one I proposed (using `await Promise.all(array.map())`) is for parallel processing, which seemed suitable to me in our case. You could also think of creating a seperate helper function like `const parallelForEach = async (array, fn) => await Promise.all(array.map(array, fn))`.

For serial processing, the most "natural" way, is using

```javascript
for (const item of array) {
  await asyncFoo(item);
}
```

`for...of` waits for the underlying promise to fulfill, before proceeding to the next item in the array. However, ESLint will complain about this, as it's forbidden in the airbnb styleguide (which we are using). They are two reasons. The first on is, that it [blocks for each item until the async operation completed](https://eslint.org/docs/latest/rules/no-await-in-loop). The second reason is, that `for...of` requires (quite big) polyfills in some browsers. We are not supporting any browsers with that syntax, so we could modify the es-lint rule (`no-restricted-syntax`). Shall I do that?

But in browser, there is the following way, which sacrifies readability and simplicitly for a significantly smaller browser bundle (as we can drop the huge polyfill):

```javascript
await array.reduce(async (acc, item) => {
    await acc; // Wait for the previous item to be processed
    await asyncFoo(item);
}, Promise.resolve()); 
```

So if you want to do serial processing, i recommend using the way described in the earlier linked [ESLint docs](https://eslint.org/docs/latest/rules/no-await-in-loop). Hope that helps and maybe you learned something :)